packages/ckeditor5-link/tests/linkimageui.js

```diff
@@ -33,9 +45,129 @@ describe( 'LinkImageUI', () => {
+			const linkPlugin = editor.plugins.get( 'LinkUI' );
+			const listenToSpy = sinon.stub( linkPlugin, 'listenTo' );
+
+			listenToSpy( viewDocument, 'click' );
+
+			viewDocument.fire( 'click' );
+
+			sinon.assert.calledOnce( listenToSpy );
+		} );
+	} );
+
+	describe( 'link toolbar UI component', () => {
+		it( 'should be registered', () => {
+			expect( linkButtonComponent ).to.be.instanceOf( ButtonView );
+		} );
+
+		describe( 'link button', () => {
+			it( 'should have a toggleable button', () => {
+				expect( linkButton.isToggleable ).to.be.true;
+			} );
+
+			it( 'should be bound to the link command', () => {
+				const command = editor.commands.get( 'link' );
+
+				command.isEnabled = true;
+				command.value = 'http://ckeditor.com';
+
+				expect( linkButton.isOn ).to.be.true;
+				expect( linkButton.isEnabled ).to.be.true;
+
+				command.isEnabled = false;
+				command.value = undefined;
+
+				expect( linkButton.isOn ).to.be.false;
+				expect( linkButton.isEnabled ).to.be.false;
+			} );
+
+			it( 'should call #_showUI upon #execute', () => {
+				const spy = testUtils.sinon.stub( editor.plugins.get( 'LinkUI' ), '_showUI' );
+
+				linkButton.fire( 'execute' );
+				sinon.assert.calledWithExactly( spy, true );
+			} );
+		} );
+	} );
+
+	describe( 'click', () => {
+		it( 'should prevent default behavior if image has "linkHref" attribute', () => {
+			setModelData( editor.model, '[<image src="" linkHref="https://example.com"></image>]' );
+
+			const img = editor.model.document.selection.getSelectedElement();
+			const data = fakeEventData();
+			const eventInfo = new EventInfo( img, 'click' );
+			const domEventDataMock = new DomEventData( viewDocument, eventInfo, data );
+
+			viewDocument.fire( 'click', domEventDataMock );
+
+			expect( data.preventDefault.called ).to.be.true;
+			expect( eventInfo.source.name ).to.equal( 'image' );
+		} );
+	} );
+
+	describe( '#_isImageLinked', () => {
+		it( 'should return "true" when selected link has "linkHref" attribute', () => {
+			setModelData( editor.model, '[<image src="" linkHref="https://example.com"></image>]' );
+
+			const img = editor.model.document.selection.getSelectedElement();
+			const spy = sinon.spy( plugin, '_isImageLinked' );
+
+			expect( spy( img ) ).to.be.true;
+		} );
+
+		it( 'should return "false" when selected link hasn\'t "linkHref" attribute', () => {
+			setModelData( editor.model, '[<image src=""></image>]' );
+
+			const img = editor.model.document.selection.getSelectedElement();
+			const spy = sinon.spy( plugin, '_isImageLinked' );
+
+			expect( spy( img ) ).to.be.false;
+		} );
+
+		it( 'should return "false" when selected element isn\'t an image', () => {
+			setModelData( editor.model, '<paragraph>foo</paragraph>' );
+
+			const el = editor.model.document.getRoot().getChild( 0 );
+			const spy = sinon.spy( plugin, '_isImageLinked' );
+
+			expect( spy( el ) ).to.be.false;
+			expect( el.name ).to.not.equal( 'image' );
+		} );
+	} );
```

Comment:
The private method should be covered by other tests. This test shouldn't be necessary as it ties the class internals. If you would in the future would like to refactor the internals (including private methods) then you'd have to rewrite this tests as well.

In other words we should test the plugin behavior or public API only.